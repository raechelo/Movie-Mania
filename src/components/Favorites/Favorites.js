import React from 'react';
import { connect } from 'react-redux';
import Card from '../../containers/Card/Card';

export const Favorites = (props) => {
  const cards = props.movies.map((m, index) => {
    if (m.favorited) {
      return <Card key={index} m={m}/>
    }
    return null;
  });
  return (
    <section className='Favorites'>
      {cards}
    </section>
  )
}

export const mapStateToProps = (state) => ({
  movies: state.movies
})

export default connect(mapStateToProps)(Favorites);