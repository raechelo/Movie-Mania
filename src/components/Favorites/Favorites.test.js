import React from 'react';
import { shallow } from 'enzyme';
import { Favorites } from './Favorites';

describe('Favorites', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <Favorites /> )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  })
})