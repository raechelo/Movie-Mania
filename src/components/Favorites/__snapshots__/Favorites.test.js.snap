// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match the snapshot with all the data passed in 1`] = `
<h2>
  Favorites
</h2>
`;
